docs/markdown/FAQ.md

```diff
@@ -399,3 +399,68 @@ the form `foo.lib` when building with MSVC, you can set the
 kwarg to `''` and the [`name_suffix:`](https://mesonbuild.com/Reference-manual.html#library)
 kwarg to `'lib'`. To get the default behaviour for each, you can either not
 specify the kwarg, or pass `[]` (an empty array) to it.
+
+## Why doesn't meson have user defined functions/macros? I **need** them!
+
+The tl;dr answer to this is that meson's design is focused on solving specific
+problems rather than providing a toolbox for developers to hobble something
+together themselves. Build systems should be quick to write and quick to
+understand, functions muddle this simplicity.
+
+The long answer is twofold:
+
+First, meson wants to provide a rich set of tools that solve specific problem
+simply out of the box. This is similar to the "batteries included" mentality of
+python. By providing tools that solve common problems in the simplest way
+possible *in* meson we are solving that problem for everyone instead of forcing
+everyone to solve that problem over and over again, often badly. Some examples
+of this are meson's dependency wrappers around various config-tool executables
+(sdl-config, llvm-config, etc). In other build systems each user of that
+dependency writes a wrapper and deals with the corner cases, in meson we handle
+them internally, everyone gets fixes and the corner cases are ironed out for
+*everyone*. Providing user defined functions or macros goes directly against
+this design goal.
+
+Second, it makes the build system easier to reason about. When you encounter
+some function call, you can refer to the reference manual to see that function
+and it's signature. Instead of spending frustrating hours trying to interpret
+some bit of m4 or follow long include paths to figure out what `function1`
+(which calls `function2`, which calls `function3`, ad infinitum), you know what
+the build system is doing. At the end of the day unless you're a meson
```

Comment:
I think the subordinate clause "unless you're a meson developer" should be set off with commas.